File: analysis/read_visits.py
# Read the synced Field Visit Log from the Google Sheet and summarise.
# pip install pandas ; set SHEET_ID (sheet must be viewable to anyone with link)
import pandas as pd
SHEET_ID = "PASTE_YOUR_SHEET_ID_HERE"
df = pd.read_csv(f"https://docs.google.com/spreadsheets/d/{SHEET_ID}/export?format=csv")
print("Total visits:", len(df), "\n")
print("Visits by field staff:\n", df["staff_name"].value_counts(), "\n")
df["week"] = pd.to_datetime(df["visit_date"]).dt.strftime("%Y-W%U")
print("Visits per staff per week:\n", df.groupby(["staff_name","week"]).size(), "\n")
print("By crop stage:\n", df["crop_stage"].value_counts(), "\n")
print("By condition:\n", df["condition"].value_counts())

File: analysis/read_visits.R
# Read the synced Field Visit Log from the Google Sheet and summarise.
# Set your Sheet ID (from its URL: /spreadsheets/d/SHEET_ID/edit) and make the
# sheet viewable to "Anyone with the link".

sheet_id <- "PASTE_YOUR_SHEET_ID_HERE"
url <- paste0("https://docs.google.com/spreadsheets/d/", sheet_id, "/export?format=csv")
df  <- read.csv(url, stringsAsFactors = FALSE, check.names = FALSE)

cat("Total visits:", nrow(df), "\n\n")

# visits by staff
cat("Visits by field staff:\n"); print(sort(table(df$staff_name), decreasing = TRUE))

# visit frequency: visits per staff per week
if (nrow(df) > 0) {
  df$week <- format(as.Date(df$visit_date), "%Y-W%U")
  cat("\nVisits per staff per week:\n")
  print(as.data.frame(table(staff = df$staff_name, week = df$week)))
}

# crop stage / condition breakdown
cat("\nBy crop stage:\n");  print(table(df$crop_stage))
cat("\nBy condition:\n");   print(table(df$condition))
